File: HumanitarianAssistance.Entities/Models/Marketing/ContractDetailsModel.cs
﻿using System;
using System.Collections.Generic;
using System.Text;

namespace HumanitarianAssistance.ViewModels.Models.Marketing
{
    public class ContractDetailsModel
    {
        public long ContractId { get; set; }
        public string ContractCode { get; set; }
        public string ClientName { get; set; }
        public long? ActivityTypeId { get; set; }
        public double UnitRate { get; set; }
        public int? CurrencyId { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public long? LanguageId { get; set; }
        public long? MediumId { get; set; }
        public long? NatureId { get; set; }
        public long? TimeCategoryId { get; set; }
        public long? MediaCategoryId { get; set; }
        public long? QualityId { get; set; }
        public bool IsCompleted { get; set; }
        public long? ClientId { get; set; }
        public long? UnitRateId { get; set; }
        public bool IsApproved { get; set; }
        public bool IsDeclined { get; set; }
        public string Type { get; set; }
        public int? Count { get; set; }
    }

    public class ApproveContractModel
    {
        public long? ContractId { get; set; }
        public string Type { get; set; }
    }

    public class FilterContractModel
    {
        public long? ContractId { get; set; }
        public string FilterType { get; set; }
        public string ClientName { get; set; }
        public long? ClientId { get; set; }
        public long? ActivityTypeId { get; set; }
        public long? CurrencyId { get; set; }
        public int? UnitRate { get; set; }
        public bool IsApproved { get; set; }
        public string YesOrNo { get; set; }
    }

    public class ContractByClient
    {
        public long? ContractId { get; set; }
        public long? ClientId { get; set; }
        public string ClientName { get; set; }
        public string ContractByClients { get; set; }
        public double UnitRate { get; set; }
        public int? CurrencyId { get; set; }
    }

    public class ContractPaginationModel
    {
        public int pageIndex { get; set; }
        public int pageSize { get; set; }
    }
}

File: HumanitarianAssistance.WebApi/NewUI/src/app/dashboard/marketing/contracts/model/contract-details.model.ts
export interface ContractDetailsModel {
    ContractId?: number;
    ContractCode?: string;
    ClientName?: string;
    UnitRate?: number;
    CurrencyId?: number;
    StartDate?: any;
    EndDate?: any;
    LanguageId?: number;
    MediumId?: number;
    NatureId?: number;
    TimeCategoryId?: number;
    MediaCategoryId?: number;
    QualityId?: number;
    IsCompleted?: boolean;
    ClientId?: number;
    ActivityTypeId?: number;
    UnitRateId?: number;
    IsApproved?: boolean;
    IsDeclined?: boolean;
    Activity?: string;
    Type?: string;
    Count?: number;
}

export interface TimeCategoryModel {
    TimeCategoryId?: number;
    TimeCategoryName?: string;
    CreatedById?: string;
    CreatedBy?: string;
    CreatedDate?: any;
    IsDeleted?: boolean;
    ModifiedBy?: string;
    ModifiedById?: string;
    ModifiedDate?: any;
}

export interface ActivityTypeModel {
    ActivityTypeId?: number;
    ActivityName?: string;
    CreatedById?: string;
    CreatedBy?: string;
    CreatedDate?: any;
    IsDeleted?: boolean;
    ModifiedBy?: string;
    ModifiedById?: string;
    ModifiedDate?: any;
}

export interface MediumModel {
    MediumId?: number;
    MediumName?: string;
    CreatedById?: string;
    CreatedBy?: string;
    CreatedDate?: any;
    IsDeleted?: boolean;
    ModifiedBy?: string;
    ModifiedById?: string;
    ModifiedDate?: any;
}

export interface NatureModel {
    NatureId?: number;
    NatureName?: string;
    CreatedById?: string;
    CreatedBy?: string;
    CreatedDate?: any;
    IsDeleted?: boolean;
    ModifiedBy?: string;
    ModifiedById?: string;
    ModifiedDate?: any;
}

export interface QualityModel {
    QualityId?: number;
    QualityName?: string;
    CreatedById?: string;
    CreatedBy?: string;
    CreatedDate?: any;
    IsDeleted?: boolean;
    ModifiedBy?: string;
    ModifiedById?: string;
    ModifiedDate?: any;
}

export interface MediaCategoryModel {
    MediaCategoryId?: number;
    CategoryName?: string;
    CreatedById?: string;
    CreatedBy?: string;
    CreatedDate?: any;
    IsDeleted?: boolean;
    ModifiedBy?: string;
    ModifiedById?: string;
    ModifiedDate?: any;
}

export interface LanguageModel {
    LanguageId?: number;
    LanguageName?: string;
    CreatedById?: string;
    CreatedBy?: string;
    CreatedDate?: any;
    IsDeleted?: boolean;
    ModifiedBy?: string;
    ModifiedById?: string;
    ModifiedDate?: any;
}

export interface CurrencyModel {
    CurrencyId?: number;
    CurrencyCode?: number;
    CurrencyRate?: number;
    CurrencyName?: string;
    CreatedById?: string;
    CreatedBy?: string;
    CreatedDate?: any;
    IsDeleted?: boolean;
    ModifiedBy?: string;
    ModifiedById?: string;
    ModifiedDate?: any;
    Status?: boolean;
    SalaryTaxFlag?: boolean;
}

export interface ApproveContractModel {
    ContractId?: number;
    Type?: string;
}
